crates/flox-rust-sdk/src/models/environment.rs

```diff
@@ -1,83 +1,858 @@
+    fn eq(&self, other: &Self) -> bool {
+        self.path == other.path
+    }
+}
+
+/// Implementation for an environment in any state
+impl<S: State> Environment<S> {
+    /// Get the owner of the environment
+    ///
+    /// **(experimental)**
+    pub fn owner(&self) -> Option<&str> {
+        self.state.owner()
+    }
+
+    /// Get the name of the environment
+    pub fn name(&self) -> &str {
+        self.state.name()
+    }
+
+    /// Turn the environment into a flake attribute,
+    /// a precise url to interact with the environment via nix
+    ///
+    /// ```
+    /// # tokio_test::block_on(async {
+    /// # use flox_rust_sdk::models::environment::{DotFloxDir, Environment, Read};
+    /// # use std::path::PathBuf;
+    /// # let tempdir = tempfile::tempdir().unwrap();
+    /// # let path = tempdir.path().canonicalize().unwrap().to_string_lossy().into_owned();
+    /// # let system = "aarch64-darwin";
+    ///
+    /// let mut dot_flox = DotFloxDir::new(&path).unwrap();
+    /// let env = dot_flox.create_env("test_env").await.unwrap();
+    ///
+    /// let flake_attribute = format!("path:{path}/.flox/test_env#.floxEnvs.{system}.default")
+    ///     .parse()
+    ///     .unwrap();
+    /// assert_eq!(env.flake_attribute(system), flake_attribute)
+    ///
+    /// # })
+    /// ```
+    pub fn flake_attribute(&self, system: impl AsRef<str>) -> FlakeAttribute {
+        let flakeref = PathRef {
+            path: self.path.clone(),
+            attributes: Default::default(),
+        }
+        .into();
+
+        let attr_path = ["", "floxEnvs", system.as_ref(), "default"]
+            .try_into()
+            .unwrap(); // validated attributes
+
+        FlakeAttribute {
+            flakeref,
+            attr_path,
+            outputs: Default::default(),
+        }
+    }
+
+    /// Path to the environment definition file
+    pub fn flox_nix_path(&self) -> PathBuf {
+        self.path.join("pkgs").join("default").join("flox.nix")
+    }
+
+    /// Build the evironment derivation using nix
+    async fn build(
         &self,
-    ) -> Result<FlakeAttribute, EnvironmentError<GenerationError<Git>, ParseInstallableError>> {
-        match self {
-            CommonEnvironment::Named(n) => n
-                .flake_attribute(Default::default())
-                .await
-                .map_err(EnvironmentError::Named),
-            CommonEnvironment::Project(p) => p.flake_attribute().map_err(EnvironmentError::Project),
+        nix: &NixCommandLine,
+        system: impl AsRef<str>,
+    ) -> Result<(), EnvironmentError2> {
+        println!("building with nix ....");
+
+        let build = Build {
+            installables: [self.flake_attribute(system).into()].into(),
+            eval: runix::arguments::eval::EvaluationArgs {
+                impure: true.into(),
+                ..Default::default()
+            },
+            build: BuildArgs {
+                // out_link: Some(self.out_link().into()),
```

Comment:
Is this a TODO? We'll need to create a link, right?

Reply:
We used to put these into a .flox ... can put them into a .flox/cache?

Reply:
The moving had some issues with the symlink so i commented this out.
We will need this just deferred to figuring out "moving directories"